Foundation/NSStream.swift

```diff
@@ -209,20 +227,32 @@ public class NSOutputStream : Stream {
         return self.init(toMemory: ())
     }
     
-    public override func propertyForKey(_ key: String) -> AnyObject? {
-        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    public override var streamError: NSError?{
+        let error = CFWriteStreamCopyError(_stream)
+        return error?._nsObject
     }
     
     public  override func setProperty(_ property: AnyObject?, forKey key: String) -> Bool {
+        guard let property = property else { return false }
         return CFWriteStreamSetProperty(_stream, key._cfObject, property)
     }
+    
+    public override func propertyForKey(_ key: String) -> AnyObject? {
+        return CFWriteStreamCopyProperty(_stream, key._cfObject)
+    }
+    
 }
 
 // Discussion of this API is ongoing for its usage of AutoreleasingUnsafeMutablePointer
 #if false
 extension Stream {
+    //Currently not tested as it is compiled out through the macro setting false
```

Comment:
sweet ill make the changes for the second version as it closer resembles the former

@phausler Also how about separate API for getting the singular stream cases and passing null/nil to CFStreamCreatePairWithSocketToHost,  so the extra is not created?
```
public class getInputStreamToHost(withName hostName: String, port: Int, inputStream: inout InputStream?){
    CFStreamCreatePairWithSocketToHost(kCFAllocatorDefault, hostName._cfObject, UInt32(port), &read, nil)
}
```